connector_salesforce/salesforce_backend/backend_model.py

```diff
@@ -0,0 +1,407 @@
+
+    You must have the full URL e.g (https://na1.salesforce.com)
+    of your instance.
+
+    There are also two means of authentication:
+    - Using username, password and security token
+    - Using IP filtering, username, password and organizationId
+    """
+
+    _name = "connector.salesforce.backend"
+    _inherit = "connector.backend"
+    _description = """Salesforce Backend"""
+    _backend_type = "salesforce"
+
+    @api.model
+    def _select_versions(self):
+        """ Available versions
+
+        Can be inherited to add custom versions.
+
+        :return: list of tuple of available versions
+        :rtype: list
+        """
+        return self._select_versions_hook()
+
+    @api.model
+    def _select_versions_hook(self):
+        """ Available versions
+
+        Can be inherited to add custom versions.
+
+        :return: list of tuple of available versions
+        :rtype: list
+        """
+        return [('15', "Winter'15")]
+
+    authentication_method = fields.Selection(
+        [
+            ('pwd_token', 'Based on User, Password, Token'),
+            ('oauth2', 'OAuth 2'),
+            ('ip_filtering', 'Based on IP Filter and OrganizationId')
+        ],
+        string='Authentication Method',
+        default='oauth2',
+    )
+
+    name = fields.Char(
+        'Name',
+        required=True
+    )
+
+    version = fields.Selection(
+        selection='_select_versions',
+        string='Version',
+        required=True
+    )
+
+    url = fields.Char(
+        'URL',
+        required=True,
+    )
+
+    username = fields.Char(
+        'User Name',
+    )
+
+    password = fields.Char(
+        'Password',
+    )
+
+    consumer_key = fields.Char(
+        'OAuth2 Consumer Key',
+    )
+
+    consumer_secret = fields.Char(
+        'OAuth2 secret',
+    )
+
+    consumer_code = fields.Char(
+        'OAuth2 client authorization code'
+    )
+    consumer_refresh_token = fields.Char(
+        'OAuth2 Refresh Token'
+    )
+    consumer_token = fields.Char(
+        'OAuth2 Token'
+    )
+    callback_url = fields.Char(
+        'Public secure URL of Odoo (HTTPS)',
+    )
+    security_token = fields.Char(
+        'Password flow Security API token',
+    )
+
+    organization_uuid = fields.Char('OrganizationId')
+
+    sandbox = fields.Boolean(
+        'Connect on sandbox instance',
+    )
+
+    @api.model
+    def _enforce_param(self, param_name):
+        """Ensure configuration parameter is set on backend record
+
+        :param param_name: name of parameter to validate
+        :type param_name: str
+
+        :return: True is parameter is set or raise an exception
+        :rtype: bool
+        """
+        self.ensure_one()
+        if not getattr(self, param_name, None):
+            f_model = self.env['ir.model.fields']
+            field = f_model.search(
+                [('model', '=', self._name),
+                 ('name', '=', param_name)],
+            )
+            if len(field) == 1:
```

Comment:
I can add a limit = 1 but the test should still be done as some fields are yet not created in ir.model.fields.table. There is a bug open on this. This issue also impact geoengine